nebula-logger/core/tests/log-management/classes/LogBatchPurger_Tests.cls

```diff
@@ -191,4 +198,132 @@ private class LogBatchPurger_Tests {
+        );
+        LoggerPlugin.setMock(mockPluginConfiguration);
+        LogBatchPurger instance = new LogBatchPurger();
+        instance.currentSObjectType = Schema.LogEntry__c.SObjectType;
+        System.assertEquals(false, ranPluginStart);
+        System.assertEquals(false, ranPluginExecute);
+        System.assertEquals(false, ranPluginFinish);
+        System.assertEquals(null, pluginConfiguration);
+        System.assertEquals(null, pluginInput);
+        List<Log__c> logsToDelete = [SELECT Id FROM Log__c];
+        System.assertNotEquals(0, logsToDelete.size());
+        MockBatchableContext mockBatchableContext = new MockBatchableContext('some_fake_job_id');
+
+        instance.execute(mockBatchableContext, logsToDelete);
+
+        LogBatchPurger.BatchPurgerInput expectedInput = new LogBatchPurger.BatchPurgerInput(mockBatchableContext, instance.currentSObjectType);
+        System.assertEquals(false, ranPluginStart);
+        System.assertEquals(true, ranPluginExecute);
+        System.assertEquals(false, ranPluginFinish);
+        System.assertEquals(mockPluginConfiguration, pluginConfiguration);
+        System.assertEquals(expectedInput.batchableContext, pluginInput.batchableContext);
+        logsToDelete = [SELECT Id FROM Log__c WHERE Id IN :logsToDelete];
+        System.assertEquals(0, logsToDelete.size(), 'All logs should have still been deleted by LogBatchPurger after running plugins: ' + logsToDelete);
+    }
+
+    @IsTest
+    static void it_should_run_plugin_finish_method() {
+        LoggerPlugin__mdt mockPluginConfiguration = new LoggerPlugin__mdt(
+            DeveloperName = 'Mock_Plugin',
+            IsEnabled__c = true,
+            BatchPurgerApexClass__c = ExampleBatchPurgerPlugin.class.getName()
+        );
+        LoggerPlugin.setMock(mockPluginConfiguration);
+        LogBatchPurger instance = new LogBatchPurger();
+        instance.currentSObjectType = Schema.LogEntry__c.SObjectType;
+        System.assertEquals(false, ranPluginStart);
+        System.assertEquals(false, ranPluginExecute);
+        System.assertEquals(false, ranPluginFinish);
+        System.assertEquals(null, pluginConfiguration);
+        System.assertEquals(null, pluginInput);
+        MockBatchableContext mockBatchableContext = new MockBatchableContext('some_fake_job_id');
+
+        instance.finish(mockBatchableContext);
+
+        LogBatchPurger.BatchPurgerInput expectedInput = new LogBatchPurger.BatchPurgerInput(mockBatchableContext, instance.currentSObjectType);
+        System.assertEquals(false, ranPluginStart);
+        System.assertEquals(false, ranPluginExecute);
+        System.assertEquals(true, ranPluginFinish);
+        System.assertEquals(mockPluginConfiguration, pluginConfiguration);
+        System.assertEquals(expectedInput.sobjectType, pluginInput.sobjectType);
+    }
+
+    public class MockBatchableContext implements Database.BatchableContext {
```

Comment:
nice!

Reply:
Yeah, I generally don't reference `Database.BatchableContext` in my batchable classes, so in tests, I typically just pass `null` for the parameter in `start()` / `execute()` / `finish()`, but having a small mock implementation seemed useful in this case

Reply:
Same, but it’s still really cool to see an actual usage